rewrite-gradle-tooling-model/model/src/main/java/org/openrewrite/gradle/marker/GradleProjectBuilder.java

```diff
@@ -94,15 +94,22 @@ static List<MavenRepository> mapRepositories(List<ArtifactRepository> repositori
                             .uri(repo.getUrl().toString())
                             .releases(true)
                             .snapshots(true);
-                    if (GradleVersion.current().compareTo(GradleVersion.version("6.6")) >= 0) {
-                        Optional.ofNullable(repo.getCredentials().getUsername()).ifPresent(builder::username);
-                        Optional.ofNullable(repo.getCredentials().getPassword()).ifPresent(builder::password);
-                    }
+                    mapAuthentication(repo, builder);
                     return builder.build();
                 })
                 .collect(toList());
     }
 
+    private static void mapAuthentication(MavenArtifactRepository repo, MavenRepository.Builder builder) {
+        try {
+            PasswordCredentials passwordCredentials = repo.getCredentials(PasswordCredentials.class);
+            Optional.ofNullable(passwordCredentials.getUsername()).ifPresent(builder::username);
+            Optional.ofNullable(passwordCredentials.getPassword()).ifPresent(builder::password);
+        } catch (IllegalArgumentException e) {
+            // We're not using password credentials
```

Comment:
Interestingly enough, `MavenRepository` hasn't been updated to support HTTP Header authentication which was quite peculiar, but I know we've seen other Gradle users using that type, so we have to guard against it here.